### ISSUE TITLE
Login failed on FreeBSD 12.3 and Ubuntu 22.04

### ISSUE THREAD

**Issue by kanryu** [neutrinolabs/xrdp#2344]
Hi, all

I've been running xrdp on FreeBSD for a long time, but after updating to xrdp-0.9.19.1 I'm having trouble logging in.

This behavior is reproduced not only on my freebsd host, but also new host on VirtualBox with FreeBSD 12.3, kde5 and xrdp installed.

As a supplement, I am connecting to the xrdp server with Windows 10 mstsc.

I would like to know how to avoid this problem.

/var/log/xrdp.log

```
[20220816-03:29:25] [INFO ] address [0.0.0.0] port [3389] mode 1
[20220816-03:29:25] [INFO ] listening to port 3389 on 0.0.0.0
[20220816-03:29:25] [INFO ] xrdp_listen_pp done
[20220816-03:29:27] [INFO ] starting xrdp with pid 717
[20220816-03:29:27] [INFO ] address [0.0.0.0] port [3389] mode 1
[20220816-03:29:27] [INFO ] listening to port 3389 on 0.0.0.0
[20220816-03:29:27] [INFO ] xrdp_listen_pp done
[20220816-03:30:16] [INFO ] Received termination signal, stopping the server accept new connections thread
[20220816-03:31:24] [INFO ] address [0.0.0.0] port [3389] mode 1
[20220816-03:31:24] [INFO ] listening to port 3389 on 0.0.0.0
[20220816-03:31:24] [INFO ] xrdp_listen_pp done
[20220816-03:31:26] [INFO ] starting xrdp with pid 729
[20220816-03:31:26] [INFO ] address [0.0.0.0] port [3389] mode 1
[20220816-03:31:26] [INFO ] listening to port 3389 on 0.0.0.0
[20220816-03:31:26] [INFO ] xrdp_listen_pp done
[20220816-03:32:42] [INFO ] Socket 12: AF_INET6 connection received from ::ffff:192.168.10.114 port 62900
[20220816-03:32:42] [INFO ] Using default X.509 certificate: /usr/local/etc/xrdp/cert.pem
[20220816-03:32:42] [INFO ] Using default X.509 key file: /usr/local/etc/xrdp/key.pem
[20220816-03:32:42] [INFO ] Security protocol: configured [SSL|RDP], requested [SSL|HYBRID|HYBRID_EX|RDP], selected [SSL]
[20220816-03:32:42] [ERROR] SSL_read: I/O error
[20220816-03:32:42] [ERROR] libxrdp_force_read: header read error
[20220816-03:32:42] [ERROR] Processing [ITU-T T.125] Connect-Initial failed
[20220816-03:32:42] [ERROR] [MCS Connection Sequence] receive connection request failed
[20220816-03:32:42] [ERROR] xrdp_sec_incoming: xrdp_mcs_incoming failed
[20220816-03:32:42] [ERROR] xrdp_rdp_incoming: xrdp_sec_incoming failed
[20220816-03:32:42] [ERROR] xrdp_process_main_loop: libxrdp_process_incoming failed
[20220816-03:32:42] [ERROR] xrdp_iso_send: trans_write_copy_s failed
[20220816-03:32:42] [ERROR] Sending [ITU T.125] DisconnectProviderUltimatum failed
[20220816-03:32:44] [INFO ] Socket 12: AF_INET6 connection received from ::ffff:192.168.10.114 port 62901
[20220816-03:32:44] [INFO ] Using default X.509 certificate: /usr/local/etc/xrdp/cert.pem
[20220816-03:32:44] [INFO ] Using default X.509 key file: /usr/local/etc/xrdp/key.pem
[20220816-03:32:44] [INFO ] Security protocol: configured [SSL|RDP], requested [SSL|HYBRID|HYBRID_EX|RDP], selected [SSL]
[20220816-03:32:44] [INFO ] Connected client computer name: DESKTOP-4IA60US
[20220816-03:32:44] [WARN ] Received [MS-RDPBCGR] TS_UD_HEADER type 0xc006 is unknown (ignored)
[20220816-03:32:44] [WARN ] Received [MS-RDPBCGR] TS_UD_HEADER type 0xc00a is unknown (ignored)
[20220816-03:32:44] [INFO ] xrdp_load_keyboard_layout: Keyboard information sent by the RDP client, keyboard_type:[0x07], keyboard_subtype:[0x02], keylayout:[0xE0010411]
[20220816-03:32:44] [INFO ] xrdp_load_keyboard_layout: model [pc105] variant [] layout [jp] options []
[20220816-03:32:44] [INFO ] TLS connection established from ::ffff:192.168.10.114 port 62901: TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384
[20220816-03:32:44] [INFO ] xrdp_caps_process_pointer: client supports new(color) cursor
[20220816-03:32:44] [INFO ] xrdp_process_offscreen_bmpcache: support level 1 cache size 5242880 MB cache entries 100
[20220816-03:32:44] [INFO ] xrdp_caps_process_codecs: nscodec, codec id 1, properties len 3
[20220816-03:32:44] [WARN ] xrdp_caps_process_codecs: unknown codec id 5
[20220816-03:32:44] [INFO ] xrdp_caps_process_codecs: RemoteFX, codec id 3, properties len 49
[20220816-03:32:44] [WARN ] Cannot find keymap file /usr/local/etc/xrdp/km-e0010411.ini
[20220816-03:32:44] [INFO ] Loading keymap file /usr/local/etc/xrdp/km-00000411.ini
[20220816-03:32:44] [WARN ] local keymap file for 0xe0010411 found and doesn't match built in keymap, using local keymap file
[20220816-03:32:56] [INFO ] connecting to sesman on 127.0.0.1:3350
[20220816-03:33:11] [ERROR] xrdp_wm_log_msg: Error connecting to sesman on 127.0.0.1:3350
[20220816-03:33:11] [INFO ] Error connecting to sesman on 127.0.0.1:3350
```

**Comment by matt335672** [comment]
Hi @kanryu 

The problem appears to be here:-

```
[20220816-03:33:11] [ERROR] xrdp_wm_log_msg: Error connecting to sesman on 127.0.0.1:3350
```

It looks like sesman isn't started, or you can't connect to it. Possible reasons:-
1) Something is preventing sesman from running
2) You've got a firewall running on the local interface lo0
3) You've disabled IPv6 on the machine.

What do you get for the following FreeBSD commands:-

```
ps auxw | grep xrdp
netstat -aLn
ifconfig lo0
```

**Comment by kanryu** [comment]
@matt335672 Thanks for writing the reply.

To confirm what you pointed out, I updated to xrdp-0.19,1 and xrdp worked fine. So my initial report may be wrong.

My host had to update its FreeBSD OS version when this problem occurred, which caused some kernel modules to stop working. So xorg wasn't working properly.

Currently, I have reviewed the settings of each module in the desktop environment, and since each is working normally, updating xrdp did not cause any problems.
